bindings/gcp/bucket/bucket.go

```diff
@@ -404,3 +411,90 @@ func (g *GCPStorage) signObject(bucket, object, ttl string) (string, error) {
 	}
 	return u, nil
 }
+
+type bulkGetPayload struct {
+	DestinationPath string `json:"destinationPath"`
+}
+
+func (g *GCPStorage) bulkGet(ctx context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata, err := g.metadata.mergeWithRequestMetadata(req)
+	if err != nil {
+		return nil, fmt.Errorf("gcp binding error while merging metadata : %w", err)
+	}
+
+	var payload bulkGetPayload
+	err = json.Unmarshal(req.Data, &payload)
+	if err != nil {
+		return nil, fmt.Errorf("gcp bucket binding error while unmarshalling bulk get payload: %w", err)
+	}
+
+	if payload.DestinationPath == "" {
+		return nil, errors.New("gcp bucket binding error: required request payload property 'destinationPath' missing")
+	}
+
+	var allObjs []*storage.ObjectAttrs
+	it := g.client.Bucket(g.metadata.Bucket).Objects(ctx, nil)
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		allObjs = append(allObjs, attrs)
+	}
+
+	var wg sync.WaitGroup
+	errCh := make(chan error, len(allObjs))
+	for _, obj := range allObjs {
+		wg.Add(1)
+		go func(object *storage.ObjectAttrs) {
+			defer wg.Done()
+			destPath := filepath.Join(payload.DestinationPath, object.Name)
+			if err := os.MkdirAll(filepath.Dir(destPath), os.ModePerm); err != nil {
+				errCh <- err
+				return
+			}
+
+			f, err := os.Create(destPath)
+			if err != nil {
+				errCh <- err
+				return
+			}
+			defer f.Close()
+
+			rc, err := g.client.Bucket(g.metadata.Bucket).Object(object.Name).NewReader(ctx)
+			if err != nil {
+				errCh <- err
+				return
+			}
+			defer rc.Close()
+
+			data, err := io.ReadAll(rc)
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			if metadata.EncodeBase64 {
+				encoded := b64.StdEncoding.EncodeToString(data)
+				data = []byte(encoded)
+			}
+
+			//nolint:gosec
+			if err := os.WriteFile(destPath, data, 0o644); err != nil {
+				errCh <- err
+				return
+			}
```

Comment:
This data should be returned, no written to a file, no? How is the data useful if written to the `daprd` filesystem.

Reply:
I got a bit confused with the ask in the issue.. As @JoshVanL points out this would write the files to daprd filesystem and not the app filesystem.

Refactored to return all objects data PTAL